polaris-react/src/components/Listbox/components/TextOption/TextOption.tsx

```diff
@@ -35,17 +36,23 @@ export const TextOption = memo(function TextOption({
     isAction && styles.isAction,
   );
 
+  const textContent = (
+    <Text as="span" variant="bodyMd">
```

Comment:
```suggestion
     <Text as="span" variant="bodyMd" fontWeight={selected ? "semibold" : "regular"}>
```

The `.selected` class font weight isn't being applied for selected TextOptions. The override in the css file can be removed too if we do the above.
![20-00-a537b-2qpgz](https://github.com/Shopify/polaris/assets/26749317/cdecacea-75c6-4960-b523-3bdd7e152392)